src/ui/handler/scroll_zoom.js

```diff
@@ -356,6 +373,38 @@ class ScrollZoomHandler {
     reset() {
         this._active = false;
     }
+
+    _addScrollZoomBlocker() {
+        if (this._map && !this._alertContainer) {
+            this._alertContainer = DOM.create('div', 'mapboxgl-scroll-zoom-blocker', this._map._container);
+
+            if (/(Mac|iPad)/i.test(window.navigator.userAgent)) {
+                this._alertContainer.textContent = this._map._getUIString('ScrollZoomBlocker.CmdMessage');
+            } else {
+                this._alertContainer.textContent = this._map._getUIString('ScrollZoomBlocker.CtrlMessage');
+            }
+
+            // dynamically set the font size of the scroll zoom blocker alert message
+            this._alertContainer.style.fontSize = `${Math.max(10, Math.min(24, Math.floor(this._el.clientWidth * 0.05)))}px`;
+        }
+    }
+
+    _isFullscreen() {
+        // scroll zoom blocker should be prevented if map is fullscreen
+        for (const control of this._map._controls) {
+            if (control instanceof FullscreenControl) return control._fullscreen;
+        }
+    }
+
+    _showBlockerAlert() {
+        if (this._alertContainer.style.visibility === 'hidden') this._alertContainer.style.visibility = 'visible';
+        this._alertContainer.classList.add('mapboxgl-scroll-zoom-blocker-show');
+
+        setTimeout(() => {
+            this._alertContainer.classList.remove('mapboxgl-scroll-zoom-blocker-show');
+        }, 200);
```

Comment:
Nit: usually it's a good practice in situations like this to save the timer id and clear it before setting to avoid the situation where many `setTimeout` callbacks are accumulated and executed in succession:

```js
clearTimeout(this._alertTimer);
this._alertTimer = setTimeout(...);
```

Reply:
Added in, thanks!